hazelcast/proxy/reliable_topic.py

```diff
@@ -1,12 +1,337 @@
+        If you don't add one, then you will be receiving the same message twice.
+
+        :return: (int), the initial sequence
+        """
+        return -1
+
+    def store_sequence(self, sequence):
+        """"
+        Informs the ReliableMessageListener that it should store the sequence.
+        This method is called before the message is processed. Can be used to
+        make a durable subscription.
+
+        :param: (int) ``sequence`` the sequence
+        """
+        pass
+
+    def is_loss_tolerant(self):
+        """
+        Checks if this ReliableMessageListener is able to deal with message loss.
+        Even though the reliable topic promises to be reliable, it can be that a
+        MessageListener is too slow. Eventually the message won't be available
+        anymore.
+
+        If the ReliableMessageListener is not loss tolerant and the topic detects
+        that there are missing messages, it will terminate the
+        ReliableMessageListener.
+
+        :return: (bool) ``True`` if the ReliableMessageListener is tolerant towards losing messages.
+        """
+        return False
+
+    def is_terminal(self):
+        """
+        Checks if the ReliableMessageListener should be terminated based on an
+        exception thrown while calling on_message.
+
+        :return: (bool) ``True` if the ReliableMessageListener should terminate itself, ``False`` if it should keep on running.
+        """
+        raise False
+
+
+class _MessageListener(object):
+    def __init__(self, uuid, proxy, to_object, listener):
+        self._id = uuid
+        self._proxy = proxy
+        self._to_object = to_object
+        self._listener = listener
+        self._cancelled_lock = threading.Lock()
+        self._cancelled = False
+        self._sequence = 0
+
+    def start(self):
+        tail_seq = self._proxy.ringbuffer.tail_sequence()
+        initial_seq = self._listener.retrieve_initial_sequence()
+        if initial_seq == -1:
+            initial_seq = tail_seq.result() + 1
+        self._sequence = initial_seq
+        self._proxy.client.reactor.add_timer(0, self._next)
+
+    def _handle_illegal_argument_error(self):
+        def on_response(res):
+            head_seq = res.result()
+            self._proxy.logger.warning("MessageListener {} on topic {} requested a too large sequence. Jumping from "
+                                       "old sequence: {} to sequence: {}".format(self._id, self._proxy.name,
+                                                                                 self._sequence,
+                                                                                 head_seq))
+            self._sequence = head_seq
+            self._next()
+
+        future = self._proxy.ringbuffer.head_sequence()
+        future.add_done_callback(on_response)
+
+    def _handle_stale_sequence_error(self):
+        def on_response(res):
+            head_seq = res.result()
+            if self._listener.is_loss_tolerant:
```

Comment:
This should call the function. `if self._listener.is_loss_tolerant:` should be `if self._listener.is_loss_tolerant():`